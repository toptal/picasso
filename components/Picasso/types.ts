import { CSSProperties } from 'react'

import { Classes } from '../styles/types'

export interface BaseProps {
  /** Classnames applied to root element */
  className?: string
  /** Style applied to root element */
  style?: CSSProperties
}

export interface JssProps {
  classes: Classes
}

export type StandardProps = BaseProps & JssProps

type Sizes = 'xsmall' | 'small' | 'medium' | 'large'

export type SizeType<T extends Sizes> = T
