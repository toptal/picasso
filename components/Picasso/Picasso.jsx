import {
  createMuiTheme,
  MuiThemeProvider,
  withStyles
} from '@material-ui/core/styles'
import CssBaseline from '@material-ui/core/CssBaseline'
import React from 'react'
import PropTypes from 'prop-types'

import {
  palette,
  layout,
  transitions,
  typography,
  spacing,
  breakpoints,
  screens
} from './config'
import FontsLoader from './FontsLoader'
import Provider from './PicassoProvider'
import globalStyles from './styles'

const picasso = {
  palette,
  layout,
  transitions,
  spacing,
  breakpoints,
  screens,
  props: {
    MuiButtonBase: {
      disableRipple: true
    },
    MuiList: {
      disablePadding: true
    },
    MuiPaper: {
      square: true
    },
    MuiOutlinedInput: {
      notched: false
    }
  },
  typography: {
    useNextVariants: true,
    inputSize: '18px',
    ...typography
  }
}

const PicassoProvider = new Provider(createMuiTheme(picasso))

const PicassoGlobalStylesProvider = withStyles(globalStyles, {
  name: 'Picasso'
})(({ children, classes }) => <div className={classes.root}>{children}</div>)

const Picasso = ({ loadFonts = true, reset = true, ...rest }) => (
  <MuiThemeProvider theme={PicassoProvider.theme}>
    {loadFonts && <FontsLoader />}
    {reset && <CssBaseline />}
    <PicassoGlobalStylesProvider {...rest} />
  </MuiThemeProvider>
)

Picasso.propTypes = {
  children: PropTypes.node.isRequired,
  loadFonts: PropTypes.bool,
  reset: PropTypes.bool
}

Picasso.defaultProps = {
  loadFonts: true,
  reset: true
}

export { PicassoProvider }
export default Picasso
