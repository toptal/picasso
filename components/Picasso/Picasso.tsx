import {
  createMuiTheme,
  MuiThemeProvider,
  withStyles
} from '@material-ui/core/styles'
import React, { FunctionComponent, ReactNode } from 'react'

import CssBaseline from '../CssBaseline'
import {
  palette,
  layout,
  transitions,
  typography,
  spacing,
  breakpoints,
  screens,
  shadows
} from './config'
import FontsLoader from './FontsLoader'
import Provider from './PicassoProvider'
import globalStyles from './styles'
import { JssProps } from './types'

const picasso = {
  palette,
  layout,
  transitions,
  spacing,
  breakpoints,
  screens,
  shadows,
  props: {
    MuiButtonBase: {
      disableRipple: true
    },
    MuiList: {
      disablePadding: true
    },
    MuiPaper: {
      square: true
    },
    MuiOutlinedInput: {
      notched: false
    }
  },
  typography: {
    useNextVariants: true,
    inputSize: '18px',
    ...typography
  }
}

const PicassoProvider = new Provider(createMuiTheme(picasso))

interface PicassoGlobalStylesProviderProps extends JssProps {
  children?: ReactNode
}

const PicassoGlobalStylesProvider = withStyles(globalStyles, {
  name: 'Picasso'
})((props: PicassoGlobalStylesProviderProps) => {
  const { classes, children } = props

  return <div className={classes.root}>{children}</div>
})

interface PicassoProps {
  children?: ReactNode
  loadFonts?: boolean
  reset?: boolean
}

const Picasso: FunctionComponent<PicassoProps> = ({
  loadFonts,
  reset,
  children
}) => (
  <MuiThemeProvider theme={PicassoProvider.theme}>
    {loadFonts && <FontsLoader />}
    {reset && <CssBaseline />}
    <PicassoGlobalStylesProvider>{children}</PicassoGlobalStylesProvider>
  </MuiThemeProvider>
)

Picasso.defaultProps = {
  loadFonts: true,
  reset: true
}

export { PicassoProvider }
export default Picasso
