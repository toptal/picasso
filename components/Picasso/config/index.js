import './theme'
export { default as palette } from './palette'
export { default as transitions } from './transitions'
export { default as typography } from './typography'
export { default as spacing } from './spacing'
export { default as breakpoints, screens } from './breakpoints'
export { default as layout } from './layout'
export { default as shadows } from './shadows'
