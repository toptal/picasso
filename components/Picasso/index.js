export { PicassoProvider } from './Picasso'
export { SizeType } from './types/SizeType'
export { default } from './Picasso'
