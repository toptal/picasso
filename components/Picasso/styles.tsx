export default ({ typography }: { typography: any }) => ({
  root: {
    height: '100%',

    '& *': {
      fontFamily: typography.fontFamily
    }
  }
})
