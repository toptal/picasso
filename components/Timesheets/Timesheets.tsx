import React from 'react'
import { withStyles } from '@material-ui/core/styles'

import Table from '../Table'
import Typography from '../Typography'
import TimesheetItem from '../TimesheetItem'
import { Classes } from '../styles/types'
import styles from './styles'

interface Props {
  classes: Classes
  /** Timesheets list */
  timesheets: {
    id: number
    startDate: string
    endDate: string
    statusMessage?: string
    isErrorStatus?: boolean
    canEdit?: boolean
    canView?: boolean
    canUnsubmit?: boolean
  }[]
  /** Callback invoked when the Edit button clicked on the timesheet record */
  onEdit: (timesheetId: number) => void
  /** Callback invoked when the View button clicked on the timesheet record */
  onView: (timesheetId: number) => void
  /** Callback invoked when the Unsubmit button clicked on the timesheet record */
  onUnsubmit: (timesheetId: number) => void
}

export const Timesheets: React.FunctionComponent<Props> = props => {
  const { timesheets, onEdit, onView, onUnsubmit } = props

  return (
    <Table>
      <Table.Head>
        <Table.Row>
          <Table.Cell>
            <Typography weight='bold'>Timesheets</Typography>
          </Table.Cell>
        </Table.Row>
      </Table.Head>
      <Table.Body>
        {timesheets.map(timesheet => (
          <Table.Row key={timesheet.id}>
            <Table.Cell>
              <TimesheetItem
                {...timesheet}
                onUnsubmit={onUnsubmit}
                onEdit={onEdit}
                onView={onView}
              />
            </Table.Cell>
          </Table.Row>
        ))}
      </Table.Body>
    </Table>
  )
}

Timesheets.defaultProps = {
  onEdit: () => {},
  onUnsubmit: () => {},
  onView: () => {}
}

Timesheets.displayName = 'Timesheets'

export default withStyles(styles)(Timesheets)
