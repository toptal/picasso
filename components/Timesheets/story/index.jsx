import PicassoBook from '../../../.storybook/components/PicassoBook'
import { Timesheets } from '../Timesheets'

const page = PicassoBook.createPage(
  'Timesheets',
  `If the Job has hourly commitment Timesheets component is showing the 
  timesheets for the work done by the talent.`,
  'Widgets'
)

page
  .addComponentDocs(Timesheets)
  .addExample('Timesheets/story/Default.example.jsx', 'Default')
  .addExample('Timesheets/story/ShowMore.example.jsx', 'Show more')
