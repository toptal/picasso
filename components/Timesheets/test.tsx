import React from 'react'
import { render, cleanup } from 'react-testing-library'

import Picasso from '../index'
import Timesheets from './index'

const renderTimesheets = (props: any) => {
  return render(
    <Picasso loadFonts={false}>
      <Timesheets {...props} />
    </Picasso>
  )
}

afterEach(cleanup)

describe('Timesheets', () => {
  const timesheets = [
    {
      id: 1,
      startDate: 'Mon, 25 Feb 2019',
      endDate: 'Sun, 10 Mar 2019',
      canEdit: true,
      canView: true
    },
    {
      id: 2,
      startDate: 'Mon, 11 Feb 2019',
      endDate: 'Sun, 24 Feb 2019',
      canUnsubmit: true,
      canEdit: true,
      canView: true
    },
    {
      id: 3,
      startDate: 'Mon, 31 Dec 2018',
      endDate: 'Sun, 13 Jan 2019',
      canView: true
    },
    {
      id: 4,
      startDate: 'Mon, 17 Dec 2018',
      endDate: 'Sun, 30 Dec 2018'
    },
    {
      id: 5,
      startDate: 'Mon, 03 Dec 2018',
      endDate: 'Sun, 16 Dec 2018',
      statusMessage: '(Submission overdue, no automatic processing available)',
      isErrorStatus: true,
      canUnsubmit: true,
      canEdit: true,
      canView: true
    },
    {
      id: 6,
      startDate: 'Mon, 22 Oct 2018',
      endDate: 'Sun, 04 Nov 2018',
      statusMessage: '(Not submitted)',
      isErrorStatus: false,
      canUnsubmit: true,
      canEdit: true,
      canView: true
    }
  ]

  test('default render', () => {
    const { container } = renderTimesheets({ timesheets })

    expect(container).toMatchSnapshot()
  })
})
