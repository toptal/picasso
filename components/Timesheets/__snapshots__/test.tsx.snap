// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timesheets default render 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            <p
              class="MuiTypography-root Typography-body Typography-bold MuiTypography-body1"
            >
              Timesheets
            </p>
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <div
              class="TimesheetItem-timesheet"
            >
              <div
                class="TimesheetItem-timesheetInfo"
              >
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  From
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Mon, 25 Feb 2019
                   
                </span>
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  till
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Sun, 10 Mar 2019
                </span>
              </div>
              <div
                class="TimesheetItem-controls"
              >
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    Edit
                  </div>
                </button>
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    View
                  </div>
                </button>
              </div>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <div
              class="TimesheetItem-timesheet"
            >
              <div
                class="TimesheetItem-timesheetInfo"
              >
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  From
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Mon, 11 Feb 2019
                   
                </span>
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  till
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Sun, 24 Feb 2019
                </span>
              </div>
              <div
                class="TimesheetItem-controls"
              >
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    Unsubmit Timesheet
                  </div>
                </button>
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    Edit
                  </div>
                </button>
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    View
                  </div>
                </button>
              </div>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <div
              class="TimesheetItem-timesheet"
            >
              <div
                class="TimesheetItem-timesheetInfo"
              >
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  From
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Mon, 31 Dec 2018
                   
                </span>
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  till
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Sun, 13 Jan 2019
                </span>
              </div>
              <div
                class="TimesheetItem-controls"
              >
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    View
                  </div>
                </button>
              </div>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <div
              class="TimesheetItem-timesheet"
            >
              <div
                class="TimesheetItem-timesheetInfo"
              >
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  From
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Mon, 17 Dec 2018
                   
                </span>
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  till
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Sun, 30 Dec 2018
                </span>
              </div>
              <div
                class="TimesheetItem-controls"
              />
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <div
              class="TimesheetItem-timesheet"
            >
              <div
                class="TimesheetItem-timesheetInfo"
              >
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  From
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Mon, 03 Dec 2018
                   
                </span>
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  till
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Sun, 16 Dec 2018
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline TimesheetItem-overdueMessage"
                >
                   
                  (Submission overdue, no automatic processing available)
                </span>
              </div>
              <div
                class="TimesheetItem-controls"
              >
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    Unsubmit Timesheet
                  </div>
                </button>
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    Edit
                  </div>
                </button>
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    View
                  </div>
                </button>
              </div>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <div
              class="TimesheetItem-timesheet"
            >
              <div
                class="TimesheetItem-timesheetInfo"
              >
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  From
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Mon, 22 Oct 2018
                   
                </span>
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                  till
                </span>
                <span
                  class="MuiTypography-root Typography-bold MuiTypography-caption MuiTypography-inline"
                >
                   
                  Sun, 04 Nov 2018
                </span>
                <span
                  class="MuiTypography-root Typography-regular MuiTypography-caption MuiTypography-inline"
                >
                   
                  (Not submitted)
                </span>
              </div>
              <div
                class="TimesheetItem-controls"
              >
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    Unsubmit Timesheet
                  </div>
                </button>
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    Edit
                  </div>
                </button>
                <button
                  class="MuiButtonBase-root Button-small Button-root"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="Button-children"
                  >
                    View
                  </div>
                </button>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
